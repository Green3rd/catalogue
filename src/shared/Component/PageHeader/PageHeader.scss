@import "../../common.scss";

$paddingX: 16px;

.PageHeader h1 {
  background-color: $mainColor;
  width: 100%;
  text-align: center;
  font-size: 1.2em;
  color: $mainTextColor;
  box-shadow: 0 0.0625rem 0.125rem 0 rgba(0, 0, 0, .2);
  display: flex;
  justify-content: center;
  margin-block-start: 0;
  margin-block-end: 0;
  // Z-index is needed because the join button has position relative
  // the button use position relative to render :before to change the color
  // can change :before to change the color in the buttin itself, to remove the position relatve and this z-index.
  z-index: 1;
}

.PageHeader {

  &__Text {
    padding: 16px 0;
    font-weight: 500;
  }

  &__BackButton {
    padding: 15px $paddingX;
    border: none;
    background-color: $mainColor;
    position: absolute;
    left: 0;

    &:hover {
      background-color: $mainColorHover;
    }

    &:focus {
      background-color: $mainColorFocus;
    }
  }

  &__BackButtonImage {
    height: 30px;

    @include for-phone-only {
      height: 22px;
    }
  }
}