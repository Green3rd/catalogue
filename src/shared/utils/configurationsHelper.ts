export const isMockMode = () => {
  return process.env.REACT_APP_USE_MOCK === "true";
};


export const isWebOpen = () => {
  return process.env.REACT_APP_OPEN_WEB === "true";
};
