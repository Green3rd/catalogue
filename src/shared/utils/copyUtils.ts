export const copyText = (text: string): boolean => {
  if (navigator.clipboard) {
    navigator.clipboard.writeText(text);
    return true;
  }
  return false;
};
