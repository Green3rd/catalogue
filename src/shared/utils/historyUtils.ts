import { routeName } from "config/routeConfig";
import * as H from "history";

export const goToHomePage = (history: H.History) => {
  history.push({
    pathname: routeName.CatalogPage,
  });
};

export const goToPreviousPage = (history: H.History) => {
  history.goBack();
};
