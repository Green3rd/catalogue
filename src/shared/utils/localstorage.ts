import { AddressFormValues } from "AddressPage/AddressForm/types";

const orderKey = "ORDER";
const randomSatangKey = "RANDOM_SATANG";
const boxKey = "BOX";
const addressKey = "ADDRESS";
const orderNumberKey = "ORDER_NUMBER";
const emailKey = "EMAIL";
const tokenKey = "TOKEN";
const isSubmit = "IS_SUBMIT";
const qrCode = "QR_CODE";
const extraCost = "EXTRA_COST";

export const setOrderToStorage = (order: number[]): void => {
  localStorage.setItem(orderKey, JSON.stringify(order));
};

export const getOrder = (): number[] => {
  const item = localStorage.getItem(orderKey);
  if (item) return JSON.parse(item);
  return Array(30).fill(0); // must be >= the highest key in the returned response.
};

export const setIsSubmitToStorage = (value: boolean): void => {
  localStorage.setItem(isSubmit, value.toString());
};

export const getIsSubmitFromStorage = (): boolean => {
  return (localStorage.getItem(isSubmit) || "false") === "true";
};

export const setRandomSatangToStorage = (randomDiscount: number): void => {
  localStorage.setItem(randomSatangKey, randomDiscount.toString());
};

export const getRandomSatangFromStorage = (): number => {
  try {
    return parseInt(localStorage.getItem(randomSatangKey) || "0") || 0;
  } catch (e: any) {
    return 0;
  }
};

export const setBoxNumberToStorage = (box: number[]): void => {
  localStorage.setItem(boxKey, JSON.stringify(box));
};

export const getBoxNumber = (): number[] => {
  const item = localStorage.getItem(boxKey);
  if (item) return JSON.parse(item);
  return Array(2).fill(0);
};

export const setAddressToStorage = (address: AddressFormValues): void => {
  localStorage.setItem(addressKey, JSON.stringify(address));
};

export const getAddress = (): AddressFormValues => {
  const item = localStorage.getItem(addressKey);
  if (item) return JSON.parse(item);
  return {
    name: "",
    number: "",
    email: "",
    address: "",
    subdistrict: "",
    district: "",
    province: "",
    zipCode: "",
    note: "",
  };
};

export const setOrderNumberToStorage = (orderNumber: string): void => {
  localStorage.setItem(orderNumberKey, orderNumber);
};

export const getOrderNumber = (): string => {
  return localStorage.getItem(orderNumberKey) || "";
};

export const setEmailToStorage = (phoneNumber: string): void => {
  localStorage.setItem(emailKey, phoneNumber);
};

export const getEmail = (): string => {
  return localStorage.getItem(emailKey) || "";
};

export const setTokenToStorage = (token: string): void => {
  localStorage.setItem(tokenKey, token);
};

export const getToken = (): string => {
  return localStorage.getItem(tokenKey) || "";
};

export const setQrCodeToStorage = (qr: string): void => {
  localStorage.setItem(qrCode, qr);
};

export const getQrCode = (): string => {
  return localStorage.getItem(qrCode) || "";
};

export const setExtraCostToStorage = (cost: number): void => {
  localStorage.setItem(extraCost, cost.toString());
};

export const getExtraCost = (): number => {
  try {
    return parseInt(localStorage.getItem(extraCost) || "0") || 0;
  } catch (e: any) {
    return 0;
  }
};
