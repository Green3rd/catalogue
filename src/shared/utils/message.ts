export const ErrorMessage = {
  cannotGetData: "เกิดข้อผิดพลาด กรุณาติดต่อ แอดมินเพจ FB: ShangarilaMoonCake",
};
