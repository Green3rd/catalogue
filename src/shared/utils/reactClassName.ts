export type Modifier = { [key: string]: (boolean | (() => boolean)) } | string[] | string;

const arrayToObject = (array: string[]) => {
    return array.reduce((acc, value) => {
        if (value) { acc[value] = true; }
        return acc;
    }, {} as any);
};

export const generateClassName = (elementClass: string, modifiers: Modifier = {}, noElementClass: boolean = false) => {
    if (!modifiers) { return elementClass; }
    const modifiersObject = typeof modifiers === 'string' ? arrayToObject(modifiers.split(' ')) :
        ((modifiers instanceof Array) ? arrayToObject(modifiers) : modifiers);
    const modifiersClasses = Object.keys(modifiersObject).reduce((acc: any, key) => {
        const modifier = modifiersObject[key];
        const value = (typeof modifier === 'function') ? modifier() : modifier;
        if (value) { acc.push(`${elementClass}--${key}`); }
        return acc;
    }, []);
    return (noElementClass ? modifiersClasses : [elementClass, ...modifiersClasses]).join(' ');
};
