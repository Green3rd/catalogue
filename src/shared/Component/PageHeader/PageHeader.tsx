import React, { MouseEventHandler } from "react";
import "./PageHeader.scss";
import arrow from "media/leftArrow.svg";

interface Props {
  text: string;
  showBackButton?: boolean;
  onBackButtonClick?: MouseEventHandler;
}

const PageHeaderComponent = (props: Props) => {
  const { text, showBackButton, onBackButtonClick } = props;
  const backButon = showBackButton && (
    <button className="PageHeader__BackButton" onClick={onBackButtonClick}>
      <img
        className="PageHeader__BackButtonImage"
        alt="back button"
        src={arrow}
      />
    </button>
  );
  return (
    <div className="PageHeader">
      {backButon}
      <h1 className="PageHeader__Text">{text}</h1>
    </div>
  );
};

const PageHeader = React.memo(PageHeaderComponent);
PageHeader.displayName = "PageHeader";
export { PageHeader };
