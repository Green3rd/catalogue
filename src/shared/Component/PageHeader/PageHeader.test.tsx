import { shallow } from "enzyme";
import { PageHeader } from "./PageHeader";
import * as ReactClassName from "shared/utils/reactClassName";
import enzymeToJson from "enzyme-to-json";

const setup = (showBackButton: boolean = true) => {
  const ownProps = {
    showBackButton,
    text: "text",
    onBackButtonClick: jest.fn(),
  };

  const props = {
    ...ownProps,
  };

  const Component = shallow(<PageHeader {...props} />);
  return { Component, props };
};

describe("<PageHeader />", () => {
  const spyReactClassName = jest.spyOn(ReactClassName, "generateClassName");
  const containerClass = "FormButton";
  beforeEach(() => {
    jest.clearAllMocks();
    spyReactClassName.mockReturnValue(containerClass);
  });

  it("should render and function correctly", () => {
    const { Component, props } = setup();

    expect(enzymeToJson(Component)).toMatchSnapshot();
    expect(props.onBackButtonClick).toBeCalledTimes(0);
    const backButton = Component.find(".PageHeader__BackButton");
    expect(backButton.exists()).toBeTruthy();
    backButton.simulate("click");
    expect(props.onBackButtonClick).toBeCalledTimes(1);
  });

  it("should not render the back button when showBackButton = false", () => {
    const { Component } = setup(false);

    const backButton = Component.find(".PageHeader__BackButton");
    expect(backButton.exists()).toBeFalsy();
  });
});
