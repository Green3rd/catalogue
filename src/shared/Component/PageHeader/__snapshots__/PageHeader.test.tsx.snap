// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> should render and function correctly 1`] = `
<div
  className="PageHeader"
>
  <button
    className="PageHeader__BackButton"
    onClick={[MockFunction]}
  >
    <img
      alt="back button"
      className="PageHeader__BackButtonImage"
      src="leftArrow.svg"
    />
  </button>
  <span
    className="PageHeader__Text"
  >
    text
  </span>
</div>
`;
